packages/cli/src/lib/commands/devices-util.ts

```diff
@@ -1,27 +1,102 @@
-import { Device } from '@smartthings/core-sdk'
+import { Device, DeviceHealth, DeviceStatus } from '@smartthings/core-sdk'
 
 import { summarizedText, TableGenerator } from '@smartthings/cli-lib'
 
 
 export type DeviceWithLocation = Device & { location?: string }
 
-export const buildTableOutput = (tableGenerator: TableGenerator, device: Device & { profileId?: string }): string => {
+export const prettyPrintAttribute = (value: unknown): string => {
+	let result = JSON.stringify(value)
+	if (result.length > 50) {
+		result = JSON.stringify(value, null, 2)
+	}
+	return result
+}
+
+export const buildStatusTableOutput = (tableGenerator: TableGenerator, data: DeviceStatus): string => {
+	let output = ''
+	if (data.components) {
+		const componentIds = Object.keys(data.components)
+		for (const componentId of componentIds) {
+			const table = tableGenerator.newOutputTable({ head: ['Capability', 'Attribute', 'Value'] })
+			if (componentIds.length > 1) {
+				output += `\n${componentId} component\n`
+			}
+			const component = data.components[componentId]
+			for (const capabilityName of Object.keys(component)) {
+				const capability = component[capabilityName]
+				for (const attributeName of Object.keys(capability)) {
+					const attribute = capability[attributeName]
+					table.push([
+						capabilityName,
+						attributeName,
+						attribute.value !== null ?
+							`${prettyPrintAttribute(attribute.value)}${attribute.unit ? ' ' + attribute.unit : ''}` : ''])
+				}
+			}
+			output += table.toString()
+			output += '\n'
+		}
+	}
+	return output
+}
+
+export const buildEmbeddedStatusTableOutput = (tableGenerator: TableGenerator, data: Device): string => {
```

Comment:
I might play with trying to merge these two functions after you merge this if that's okay. (I'm thinking maybe a single generic function and then two glue functions that call it, assuming I can figure out how to make the types work.)